### PR TITLE
purger: fix nil value in column range for BatchWrite query

### PR DESCRIPTION
**Pull request by jakubgs**
Fixes following error:
```
msg="error adding delete request to the store" err="Invalid null value in condition for column range"
```
Resolves: https://github.com/cortexproject/cortex/issues/3237

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -71,6 +71,7 @@
 * [BUGFIX] Frontend, Query-scheduler: allow querier to notify about shutdown without providing any authentication. #4066
 * [BUGFIX] Querier: fixed race condition causing queries to fail right after querier startup with the "empty ring" error. #4068
 * [BUGFIX] Compactor: Increment `cortex_compactor_runs_failed_total` if compactor failed compact a single tenant. #4094
+* [BUGFIX] Purger: fix `Invalid null value in condition for column range` caused by `nil` value in range for WriteBatch query. #4128
```
Comment (pracucci) [inline CHANGELOG.md:74]: We've just cut a new Cortex version. Can you rebase `master` and move this to the top, under "master / unreleased"?

**Review by pracucci — COMMENTED** [review]
@sandeepsukhani Could you review this, please?

File: CHANGELOG.md (hunk 1 above)
Comment (jakubgs) [reply CHANGELOG.md]: Done and done.

**Review by sandeepsukhani — APPROVED** [review]
LGTM

**Comment by jakubgs** [comment]
I don't see how the test failures are related to my changes:
```
--- FAIL: TestAlertmanagerSharding (57.27s)
    --- PASS: TestAlertmanagerSharding/bucket_alertstore,_RF_=_2 (14.24s)
    --- PASS: TestAlertmanagerSharding/legacy_alertstore,_RF_=_3 (14.54s)
    --- FAIL: TestAlertmanagerSharding/bucket_alertstore,_RF_=_3 (16.24s)
    --- PASS: TestAlertmanagerSharding/legacy_alertstore,_RF_=_2 (12.25s)
```
Is it just a fluke?

**Review by pracucci — APPROVED** [review]

**Review by pstibrany — APPROVED** [review]
Thanks. Can you please fix conflict in CHANGELOG?

**Comment by jakubgs** [comment]
Will somebody merge this, or will I have to rebase `CHANGELOG.md` changes for the third time?

**Comment by pstibrany** [comment]
> Will somebody merge this, or will I have to rebase `CHANGELOG.md` changes for the third time?

PR is now set to merge after CI checks pass. Thank you.

**Comment by jakubgs** [comment]
Yeah, but CI checks fail on random stuff totally unrelated to my changes, like:
```
--- FAIL: TestAlertmanager_StateReplicationWithSharding_InitialSyncFromPeers (3.88s)
    --- FAIL: TestAlertmanager_StateReplicationWithSharding_InitialSyncFromPeers/RF_=_2 (1.49s)
        multitenant_test.go:1617: 
            	Error Trace:	multitenant_test.go:1617
            	Error:      	Not equal: 
            	            	expected: 1
            	            	actual  : 0
            	Test:       	TestAlertmanager_StateReplicationWithSharding_InitialSyncFromPeers/RF_=_2
FAIL
FAIL	github.com/cortexproject/cortex/pkg/alertmanager	111.319s
```

**Pull request merged**